delivery_transsmart/models/res_config.py

```diff
@@ -108,47 +108,87 @@ def get_transsmart_service(self):
     def get_transsmart_carrier_tag(self):
         return self.env['ir.model.data'].get_object('delivery_transsmart', 'res_partner_category_transsmart_carrier')        
 
+    def _get_odata_filter(self, transsmart_ids):
+        """
+        assembling an odata $filter
+        http://www.odata.org/documentation/
+        odata-version-2-0/uri-conventions/
+
+        Gets all the records that their Id is not in transsmart_ids
+        """
+        _filter = ''
+        for index, code in enumerate(transsmart_ids):
+            if index < len(transsmart_ids) - 1:
+                _filter += 'Id ne {} and '.format(transsmart_ids[index])
+            else:
+                _filter += 'Id ne {}'.format(transsmart_ids[index])
+        return _filter
+
     @api.multi
     def sync_transsmart_models(self):
```

Comment:
I see a lot of nearly identical code repeating. Like get transsmart data for some model. Filter it. Loop over it. Write what we already have, create what we do not have. It should really be possible to make some general method and call it with the arguments (or maybe a dictionary of info) to handle what is different in those loops.